FormlyMapper/FormlyField.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Linio\DynamicFormBundle\FormlyMapper;
+
+abstract class FormlyField implements FormlyFieldInterface
+{
+    /**
+     * @var array
+     */
+    protected $fieldConfiguration;
+
+    /**
+     * @var array
+     */
+    protected $formlyFieldConfiguration;
+
+    /**
+     * @param array $fieldConfiguration
+     */
+    public function setFieldConfiguration(array $fieldConfiguration)
+    {
+        $this->fieldConfiguration = $fieldConfiguration;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFormlyFieldConfiguration()
```

Comment:
This is still a getter that is changing state. I would probably create a protected method called `buildConfiguration()` that has this logic, and then calls `$this->buildFieldTypeConfiguration()`. `buildFieldTypeConfiguration()` would also exist in this abstract class, and would do nothing by default. This way you don't have to have it in every child class for no reason.